pkg/imrpc/proxy.proto

```diff
@@ -164,6 +164,7 @@ message EngineSnapshotBackupStatusRequest {
 
 	string backup_name = 2;
 	string replica_address = 3;
+	string replica_name = 4;
```

Comment:
Not sure if it's better to add this to `ControllerReplica`. So far the SPDK engine/controller will tell the replica address along with the name.

Reply:
In general, I think it WOULD be better. However, as you are alluding to, it requires us to refactor the in-memory data structures the longhorn-engine controller server uses to keep track of replicas (currently, the v1 engine itself has no concept of the name of a replica). I am glad the v2 engine is enhanced to understand replica name in its initial release, as this will prevent a number of problems. However, the v1 refactoring required for this suggestion is out-of-scope in the existing LEP: https://github.com/longhorn/longhorn/blob/master/enhancements/20230420-engine-identity-validation.md#non-goals.